### DISCUSSION TITLE
Refine tokens

### DISCUSSION THREAD

**Discussion by basham** [indiana-university/rivet-source#553] [Feature and enhancement requests]
There's a number of ways Rivet tokens can be refined. I want to discuss these proposals:

1. Define tokens with CSS variables. (Also see, #422.)
2. Define a `0.25rem` (`4px`) grid.
3. Use a numbering system rather than or in addition to t-shirt sizing for any size-related token.
4. Add line-height tokens.
5. Define recommended font-size and line-height pairings.
6. Simplify the color scale names.

These are all outlined in my [Rivet tokens proposal (CodePen)](https://codepen.io/basham/pen/oNBVdVg).

**Comment by basham** [comment]
Exemplar: The [Open Props project](https://open-props.style/) is essentially just a bunch of CSS variables that you can use to composite whatever you want. It's interesting to show how widely useful it can be, if they were an option.

**Reply by levimcg** [reply]
I saw this project a couple of times on Twitter the other day, but hadn't really had time to check it out.

re: Rivet CSS variables. We are currently shipping a set of CSS variables in the npm package—also [available via unpkg](https://unpkg.com/browse/rivet-core@2.0.0-alpha.8/tokens/css/variables.css)

This is something we still need to document before the production 2.0.0 release next year, but they technically are available. The `tokens/*` folder gets generate using [Style Dictionary](https://amzn.github.io/style-dictionary/#/) at publish time. As you layout in your above proposal, there are definitely more things that we can _tokenize_ and manage via the [source `tokens/*.json` files](https://github.com/indiana-university/rivet-source/tree/2.0.0-master/src/tokens) which are used by Style Dictionary to generate Sass, CSS variables, etc.
